rat/compiler.py

```diff
@@ -498,207 +577,155 @@ def _prepare_for_scans(self):
                                 msg = f"Recursively assigning {symbol_key} requires a shifted subscript on the right hand side reference"
                                 raise CompileError(msg, self.model_code_string, symbol.line_index, symbol.column_index)
 
-                            shifts = symbol.subscript.shifts
+                            shifts = [shift.value for shift in symbol.subscript.shifts]
 
-                            if sum(shift is not None for shift in shifts) != 1:
+                            if sum(shift != 0 for shift in shifts) != 1:
                                 msg = "Exactly one (no more, no less) subscripts can be shifted in a recursively assigned variable"
                                 raise CompileError(msg, self.model_code_string, symbol.line_index, symbol.column_index)
 
-                            if shifts[-1] is None:
```

Comment:
I changed the "rightmost subscript must have positive shift" condition for assignment into "a single subscript must have a positive shift".

Reply:
Ah, the reason it's the rightmost is performance. To do the scans we have to save in the rolling buffer so the previous element is in fact the previous element in that buffer. If we allow other positions then the rolling buffer logic is slower (and also not sure it would work with ragged array type stuff)

Reply:
Ohhh I see. I'll revert that back